next/pages/[...slug].tsx

```diff
@@ -99,13 +99,14 @@ export const getStaticProps: GetStaticProps<PageProps> = async (context) => {
   // Are we in Next.js preview mode?
   const isPreview = context.preview || false;
 
-  // Get the page data with Graphql:
-  const data = await drupal.doGraphQlRequest(
-    GET_ENTITY_AT_DRUPAL_PATH,
-    variables,
-    // We want to use authentication only if this is a preview request:
-    isPreview ? true : false,
-  );
+  // Get the page data with Graphql.
+  // We want to use a different client if we are in preview mode:
+  const data = isPreview
+    ? await drupalPreviewer.doGraphQlRequest(
+        GET_ENTITY_AT_DRUPAL_PATH,
+        variables,
+      )
+    : await drupal.doGraphQlRequest(GET_ENTITY_AT_DRUPAL_PATH, variables);
 
   // If the data contains a RedirectResponse, we redirect to the path:
   const redirect = extractRedirectFromRouteQueryResult(data);
```

Comment:
here we can switch to a different client based on the fact that we are in preview or not. Should still be readable, I think?